### DISCUSSION TITLE
VocabParallelClassifier1D中设置是否有问题？

### DISCUSSION THREAD

**Discussion by yhcc** [hpcaitech/ColossalAI#2256] [Community | Q&A]
我发现在VocabParallelClassifier1D中会设置
https://github.com/hpcaitech/ColossalAI/blob/c8c79102f02f66f7fb9576d51653843a8bf8ce6f/colossalai/nn/layer/parallel_1d/layers.py#L345
同时计算loss的时候，貌似会用到这个环境变量
https://github.com/hpcaitech/ColossalAI/blob/c8c79102f02f66f7fb9576d51653843a8bf8ce6f/colossalai/nn/loss/__init__.py#L32

但是VocabParallelClassifier1D有一个这个参数https://github.com/hpcaitech/ColossalAI/blob/c8c79102f02f66f7fb9576d51653843a8bf8ce6f/colossalai/nn/layer/parallel_1d/layers.py#L316 ，如果设置了这个微True的话，loss就不需要并行的计算了，这里感觉会出现bug？

**Comment by kurisusnowdeng** [comment]
非常感谢您的反馈。应该是确实存在这个问题，gather_output本来主要是用在模型在transformer之后是连续两个linear去输出logits时（比如bert）的倒数第二个linear层，保证它的输出的tensor是不并行的（本身1D的话此处的column linear必然会输出并行的tensor）。我们会尽快优化一下相关接口的易读性，也欢迎您提出建议，谢谢。
